src/Symfony/Component/HttpKernel/Profiler/Profiler.php

```diff
@@ -215,8 +223,8 @@ public function collect(Request $request, Response $response, \Exception $except
         foreach ($this->collectors as $collector) {
             $collector->collect($request, $response, $exception);
 
-            // forces collectors to become "read/only" (they loose their object dependencies)
-            $profile->addCollector(unserialize(serialize($collector)));
+            // we need to clone for sub-requests
+            $profile->addCollector(clone $collector);
```

Comment:
I know this change was a long time ago, but I was just curious if you remembered why this change was needed. Just took a dive into profiling the profiler and it seems like a large amount of time is spent here, wondering what broke to cause this change.